components/omega/src/ocn/Eos.h

```diff
@@ -0,0 +1,257 @@
+#ifndef OMEGA_EOS_H
+#define OMEGA_EOS_H
+//===-- ocn/Eos.h - Equation of State --------------------*- C++ -*-===//
+//
+/// \file
+/// \brief Contains functors for calculating specific volume
+///
+/// This header defines functors to be called by the time-stepping scheme
+/// to calculate the specific volume based on the choice of EOS
+//
+//===----------------------------------------------------------------------===//
+
+#include "AuxiliaryState.h"
+#include "Config.h"
+#include "EosConstants.h"
+#include "HorzMesh.h"
+#include "MachEnv.h"
+#include "OmegaKokkos.h"
+#include "TimeMgr.h"
+#include <string>
+
+namespace OMEGA {
+
+enum class EosType {
+   Linear,       /// Linear equation of state
+   TEOS10Poly75t /// Roquet et al. 2015 75 term expansion
+};
+
+//
+/// TEOS10 75-term polynomial
+class TEOS10Poly75t {
+ public:
+   Array2DReal specVolPcoeffs;
+
+   /// constructor declaration
+   TEOS10Poly75t(int NVertLevels);
+
+   KOKKOS_FUNCTION void operator()(const Array2DReal &SpecVol, I4 ICell,
```

Comment:
If SpecVol is owned by the Eos class, we should return a pointer to that internal variable as the return value for the function.

Reply:
Also, I noticed it is declared const in the arg list. As a return value, it should not be const.

Reply:
Actually on second thought, Kokkos array copies are just copying the pointer anyway, so it's fine to leave as is. Sorry for the confusion.